ui/de-lib/src/main/java/org/iplantc/de/client/util/OntologyUtil.java

```diff
@@ -101,4 +125,74 @@ public AvuList convertHierarchiesToAvus(List<OntologyHierarchy> selectedHierarch
         return avuListBean;
     }
 
+    public List<String> getPathList(OntologyHierarchy hierarchy) {
+        List<String> pathList = Lists.newArrayList();
+        if (hierarchy != null) {
+            String tag = getHierarchyPathTag(hierarchy);
+            pathList = Arrays.asList(tag.split("/"));
+        }
+        return pathList;
+    }
+
+    public List<List<String>> getAllPathsList(List<OntologyHierarchy> hierarchies) {
+        List<List<String>> pathList = Lists.newArrayList();
+        if (hierarchies == null || hierarchies.size() == 0){
+            return pathList;
+        }
+
+        Collections.sort(hierarchies, new OntologyHierarchyNameComparator());
+        for (OntologyHierarchy hierarchy : hierarchies) {
+            pathList.add(getPathList(hierarchy));
+        }
+        return pathList;
+    }
+
+    /**
+     * Given a hierarchy it will return that hierarchy's key.
+     * The key is actually a tag which is a "/" separated string of the path from the root to this node.
+     * In order to achieve this, any time a key is defined for a node, its children must get updated with
+     * the parents key in a separate parent tag.
+     * @param hierarchy
+     * @return
+     */
+    public String treeStoreModelKeyProvider(OntologyHierarchy hierarchy) {
+        String key = getHierarchyPathTag(hierarchy);
+        setChildrenParentTag(key, hierarchy);
+        return key;
+    }
```

Comment:
I'm not sure these method names are very intuitive for what they accomplish.
Perhaps a method name like `getOrCreateHierarchyPathTag` would be better?
Also, could the `setChildrenParentTag` method only be called if the model's tag needed to be created, instead of every time the key is fetched?